### PR TITLE
[DO NOT MERGE] Build changes (disable a ton of plugins) to enable building with sbt 1.0

### PR DESCRIPTION
**Pull request by ktoso**
DO NOT MERGE

Keeping it around so we can keep looking at what's missing for us to really upgrade to sbt 1.0

Main parts still to do now:
* [x] unidoc (https://github.com/akka/akka/pull/23672)
* [x] protobuf (#23730)
* [x] release (also https://github.com/akka/akka/pull/23672)
* [x] paradox
* [x] formatting
* [ ] sbt-dependency-graph (for `validatePullRequest`) #23735
* ?

### CODE REVIEW

File: project/scripts/release [hunk 1]
```diff
@@ -76,7 +74,7 @@
 # https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide
 
 # defaults
-declare -r default_server="akkarepo@gustav.akka.io"
+declare -r default_server="akkarepo@repo.akka.io"
```
Comment (raboof) [inline project/scripts/release]: This can be updated to 'gustav'

Comment (ktoso) [reply project/scripts/release]: thanks :)

**Comment by raboof** [comment]
sbt1 support in akka-paradox is in https://github.com/akka/akka-paradox/pull/14

File: akka-actor/build.sbt [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import akka.{ AkkaBuild, Formatting, OSGi, Dependencies, Version }
+import akka.{ AkkaBuild, Formatting, OSGi, Dependencies, VersionGenerator }
 
 AkkaBuild.defaultSettings
 Formatting.formatSettings
 OSGi.actor
 Dependencies.actor
```
Comment (ktoso) [inline akka-actor/build.sbt]: That’s weird - conflicted with what, do you know? Just the class that we have callled like that in class path...?

Comment (raboof) [reply akka-actor/build.sbt]: Right: to the generated `akka.Version` that's pulled in as a dependency of sbt-web

**Comment by ktoso** [comment]
Huh, very weird - thanks for spotting!

—
Konrad `kto.so` Malawski
Akka <http://akka.io> @ Lightbend <http://lightbend.com>

On 20 September 2017 at 23:46:45, Arnout Engelen (notifications@github.com)
wrote:

*@raboof* commented on this pull request.
------------------------------

In akka-actor/build.sbt
<https://github.com/akka/akka/pull/23651#discussion_r139991482>:

> @@ -1,10 +1,10 @@
-import akka.{ AkkaBuild, Formatting, OSGi, Dependencies, Version }
+import akka.{ AkkaBuild, Formatting, OSGi, Dependencies, VersionGenerator }

 AkkaBuild.defaultSettings
 Formatting.formatSettings
 OSGi.actor
 Dependencies.actor

Right: to the generated akka.Version that's pulled in as a dependency of
sbt-web

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/akka/akka/pull/23651#discussion_r139991482>, or mute
the thread

File: akka-stream/src/main/java-jdk9-only/akka/stream/javadsl/JavaFlowSupport.java [hunk 3]
```diff
@@ -102,21 +102,26 @@ private Flow() {
...
-
-            return new java.util.concurrent.Flow.Processor<In, Out>() {
-              @Override public void onError(Throwable t) { sub.onError(t); }
-              @Override public void onSubscribe(java.util.concurrent.Flow.Subscription s) { sub.onSubscribe(s); }
-              @Override public void onComplete() { sub.onComplete(); }
-              @Override public void onNext(In t) { sub.onNext(t); }
-              @Override public void subscribe(java.util.concurrent.Flow.Subscriber<? super Out> s) { pub.subscribe(s); }
-            };
-          });
+          .mapMaterializedValue(pair -> new WrappingProcessor<In, Out>(pair.first(), pair.second()));
```
Comment (jrudolph) [inline akka-stream/src/main/java-jdk9-only/akka/stream/javadsl/JavaFlowSupport.java]: :+1 good idea

Comment (jrudolph) [reply akka-stream/src/main/java-jdk9-only/akka/stream/javadsl/JavaFlowSupport.java]: Luckily it seems we could rather easily avoid those code structures

File: project/scripts/release [hunk 4]
```diff
@@ -168,8 +168,6 @@ fi
 declare -r version=$1
 declare -r publish_path="${release_server}:${release_path}"
 
-[[ `java -version 2>&1 | grep -E "java version|openjdk version" | cut -d ' ' -f3 | cut -d '.' -f2` -eq 8 ]] || fail "Java version is not 1.8"
-
```
Comment (ktoso) [inline project/scripts/release]: Should we fail if it is not 9 though instead?

File: project/scripts/release [hunk 5]
```diff
@@ -159,7 +159,7 @@ if [ $# != "1" ]; then
   fail "A release version must be specified"
 fi
 
-declare -r java_version="$(java -version 2>&1 | awk -F '"' '/version/ {print $2}')"
+declare -r java_version="$(java -version 2>&1 | awk -F '"' '/version/ {print $2}' | sed -e 's/-Debian//')"
```
Comment (ktoso) [inline project/scripts/release]: ok :)

File: project/scripts/release [hunk 6]
```diff
@@ -310,7 +310,8 @@ if [ ! $no_mima ] && [ $dry_run ]; then
   echodry "Finished migration manager report"
 fi
 
-try sbt whitesourceCheckPolicies
+# FIXME whitesource does not work with jdk9, https://github.com/whitesource/agents/issues/13
+# try sbt whitesourceCheckPolicies
```
Comment (ktoso) [inline project/scripts/release]: oh ok

**Comment by lightbend-cla-validator** [comment]
Hi @novacekm,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: project/scripts/release (hunk 4 above)
Comment (raboof) [reply project/scripts/release]: Yes, and we do - it's a couple of lines up IIRC

**Comment by novacekm** [comment]
CLA signed

**Pull request closed without merging**